File: app/src/main/java/com/kazakago/cleanarchitecture/di/component/ApplicationComponent.java
package com.kazakago.cleanarchitecture.di.component;

import com.kazakago.cleanarchitecture.di.module.ApplicationModule;
import com.kazakago.cleanarchitecture.di.module.DataModule;
import com.kazakago.cleanarchitecture.di.module.DomainModule;
import com.kazakago.cleanarchitecture.presentation.presenter.fragment.AboutFragmentViewModel;
import com.kazakago.cleanarchitecture.presentation.presenter.fragment.MainFragmentViewModel;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {ApplicationModule.class, DomainModule.class, DataModule.class})
public interface ApplicationComponent {

    void inject(MainFragmentViewModel mainFragmentViewModel);

    void inject(AboutFragmentViewModel aboutFragmentViewModel);

}

File: app/src/main/java/com/kazakago/cleanarchitecture/di/component/ApplicationComponent.kt
package com.kazakago.cleanarchitecture.di.component

import com.kazakago.cleanarchitecture.di.module.ApplicationModule
import com.kazakago.cleanarchitecture.di.module.DataModule
import com.kazakago.cleanarchitecture.di.module.DomainModule
import com.kazakago.cleanarchitecture.di.module.WebModule
import com.kazakago.cleanarchitecture.presentation.presenter.fragment.AboutFragmentViewModel
import com.kazakago.cleanarchitecture.presentation.presenter.fragment.MainFragmentViewModel
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(ApplicationModule::class, DomainModule::class, DataModule::class, WebModule::class))
interface ApplicationComponent {

    fun inject(mainFragmentViewModel: MainFragmentViewModel)

    fun inject(aboutFragmentViewModel: AboutFragmentViewModel)

}
